recipes/protobuf/all/conanfile.py

```diff
@@ -190,11 +253,17 @@ def package(self):
         cmake = CMake(self)
         cmake.install()
         rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
-        os.unlink(os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-config-version.cmake"))
-        os.unlink(os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-targets.cmake"))
-        os.unlink(os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-targets-{}.cmake".format(str(self.settings.build_type).lower())))
-        rename(self, os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-config.cmake"),
-                     os.path.join(self.package_folder, self._cmake_install_base_path, "protobuf-generate.cmake"))
+        install_dir = os.path.join(self.package_folder, self._cmake_install_base_path)
+        os.unlink(os.path.join(install_dir, "protobuf-config-version.cmake"))
+        os.unlink(os.path.join(install_dir, "protobuf-targets.cmake"))
+        os.unlink(os.path.join(install_dir, "protobuf-targets-{}.cmake".format(str(self.settings.build_type).lower())))
+
+        config_path = os.path.join(install_dir, "protobuf-config.cmake")
+        if Version(self.version) < "3.22.0":
+            rename(self, config_path, os.path.join(install_dir, "protobuf-generate.cmake"))
+        else:
+            os.unlink(config_path)
+            rmdir(self, os.path.join(install_dir, "../utf8_range"))
```

Comment:
Also here, what's in the utf8_range folder? Why is it being removed?

Reply:
cmake config files for utf8_range. They were checked by the hook.